lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -0,0 +1,233 @@
+        empty_response_payload: Dict[AnyStr, Any] = None,
+    ):
+        self.pbench_endpoint = pbench_endpoint
+        self.elastic_endpoint = elastic_endpoint
+        self.payload = payload
+        self.bad_date_payload = bad_date_payload
+        self.error_payload = error_payload
+        self.empty_response_payload = empty_response_payload
+
+    def build_index(self, server_config, dates):
+        """
+        Build the index list for query
+
+        Args:
+            dates (iterable): list of date strings
+        """
+        idx = server_config.get("Indexing", "index_prefix") + ".v6.run-data."
+        index = "/"
+        for d in dates:
+            index += f"{idx}{d},"
+        return index
+
+    def test_non_accessible_user_data(self, client, server_config, pbench_token):
+        """
+        Test behavior when Authorization header does not have access to other user's data
+        """
+        # The pbench_token fixture logs in as user "drb"
+        # Trying to access the data belong to the user "pp"
+        if not self.payload.get("user", None):
+            pytest.skip("skipping non accessible user data test")
+        self.payload["user"] = "pp"
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+    @pytest.mark.parametrize(
+        "user", ("drb", "pp"),
+    )
+    def test_accessing_user_data_with_invalid_token(
+        self, client, server_config, pbench_token, user
+    ):
+        """
+        Test behavior when expired Authorization header provided
+        """
+        if not self.payload.get("user", None):
+            pytest.skip("skipping accessing user data with invalid token test")
+        # valid token logout
+        response = client.post(
+            f"{server_config.rest_uri}/logout",
+            headers=dict(Authorization="Bearer " + pbench_token),
+        )
+        assert response.status_code == HTTPStatus.OK
+        self.payload["user"] = user
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+    def test_missing_json_object(self, client, server_config, pbench_token):
+        """
+        Test behavior when no JSON payload is given
+        """
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        assert response.json.get("message") == "Invalid request payload"
+
+    def missing_keys(self, client, server_config, keys, user_ok, pbench_token):
```

Comment:
Oh, boo.

So, I still think it's worth storing the class object as an attribute of the test class (which will DRY out the code more), but, yeah, passing in the `keys` value from the PyTest parameterization is probably the best approach, rather than trying to find some way to hang that off the test class as well.  Pity.

Reply:
Yes indeed, I implemented this in the current commit

Reply:
That would, indeed, be most cool!